### PR TITLE
Adding beam stop arm masking for all ISIS SANS instruments

### LINKED ISSUE #22266: SANS V2 GUI arm masking not working for LOQ

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
The beam stop arm masking in the new SANS Gui was only enabled for SANS2D. This enables it for LOQ, LARMOR and ZOOM as well.

**To test:**

- Get the data from the following folder
[loqdemo.zip](https://github.com/mantidproject/mantid/files/1895126/loqdemo.zip)

- Open up the SANS GUI V2. Load MaskFile.txt as the user file and batch.csv as the batch file.
- Go to settings->Masks and click display mask confirm that a beam stop arm is masked out.

Fixes #22266 . 

**Release Notes** 

Have updated release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/SANS/sans/algorithm_detail/mask_workspace.py [hunk 1]
```diff
@@ -299,7 +299,7 @@ def mask_angle(mask_info, workspace):
     return workspace
 
 
-def mask_beam_stop(mask_info, workspace, instrument):
+def mask_beam_stop(mask_info, workspace, instrument, detector_names):
     """
     The beam stop is being masked here. Note that this is only implemented for SANS2D
```
Comment (KarlPalmen) [inline scripts/SANS/sans/algorithm_detail/mask_workspace.py]: Won't this comment be out of date?

**Review by KarlPalmen — CHANGES_REQUESTED** [review]

**Comment by KarlPalmen** [comment]
I ran the test as described and I got a picture of the instrument after setting the instrument to LOQ in the first tab. It showed an arm that was not present when I did the same with an older build of Mantid.

I caused Mantid to crash when I attempted to display mask a second time. This also applied to the older build, so is not part of this issue.  

I found no reference to the arm in the documentation.
I'll approve, once the comment I referred to has been dealt with.

**Comment by KarlPalmen** [comment]
I've reported the crash as #22288.

**Comment by Mantid-Matthew** [comment]
The relevant documentation is here https://www.mantidproject.org/SANS_User_File_Commands#MASK . I think it is currently out of date and unhelpful however. I'm talking to the instrument scientists about what to change it to and will update it once I get a response.

**Review by KarlPalmen — APPROVED** [review]

**Comment by KarlPalmen** [comment]
I looked at http://docs.mantidproject.org/nightly/interfaces/ISIS%20SANS%20v2.html .

**Comment by Mantid-Matthew** [comment]
I think that mention of the beam arm is missed out there actually as it should probably be in the list of the kind of masks. I'll update it in this pull request otherwise it will slip through the cracks.

**Comment by KarlPalmen** [comment]
I've put the in progress label on, to stop anyone from mergin this yet.

File: docs/source/interfaces/ISIS SANS v2.rst [hunk 2]
```diff
@@ -284,7 +284,7 @@ Masking information
 
 The masking table shows detailed information about the masks which will be applied.
 These masks include bin masks, cylinder masks, mask files, spectrum masks, angle masks
-and masks for the beam stop. If as mask is applied only to a particular detector
+and line masks for the beam stop arm. If a mask is applied only to a particular detector
 then this will be shown in the masking table. Note that data needs to be specified
```
Comment (KarlPalmen) [inline docs/source/interfaces/ISIS SANS v2.rst]: I determined there was no reference to the beam stop arm, by searching for 'arm', which only occurred in LARMOUR.

**Pull request merged**